_plugins/api.rb

```diff
@@ -85,6 +93,38 @@ def self.generate_locations_endpoint(site)
         'Index of team members by location code', data)
     end
 
+    # Based off of
+    # https://github.com/jekyll/jekyll/blob/v2.5.2/test/test_page.rb#L10-L13
+    # Modifies `page`
+    def self.do_render(site, page)
+      layout = Jekyll::Layout.new(site, '_layouts', 'simple.html')
+      layouts = Hash.new(layout)
+      page.render(layouts, site.site_payload)
+    end
+
+    def self.pages(site)
+      site.pages.select {|page| %w(.html .md).include?(page.ext) }
+    end
+
+    def self.get_output(site, page)
+      do_render(site, page)
+      filterer = Filterer.new
+      filterer.condense(filterer.strip_html(page.content))
+    end
```

Comment:
> I was assuming that the normal Jekyll site generation would take place, then the indexing would happen afterwards.

That could work. Ideally it wouldn't require any monkey-patching or extra build step, but that might be good-enough-for-now.

> There is a `page.render()` you can use in a pinch

Yep, using that a few lines above... is there a step or option or something I'm missing?

> it'll slow down the build because those pages will be rendered twice.

I'm ok with that for an MVP, but good to know!